wsd/RequestVettingStation.cpp

```diff
@@ -261,9 +261,13 @@ void RequestVettingStation::handleRequest(const std::string& id,
                         }
                     }
                     else if (_checkFileInfo == nullptr ||
-                             _checkFileInfo->state() == CheckFileInfo::State::None)
+                             _checkFileInfo->state() != CheckFileInfo::State::Pass)
```

Comment:
Hmm - that's quite a big change; to double check the TimedOut state != Pass - and will now not go to the  // E.g. Timeout path (?)

Reply:
This function is called when cool.html is loaded and the WebSocket is created. By this point, `_checkFileInfo` was already working in the background getting CheckFileInfo.

If we had timed out, I want to try again. Hence the change to `!= Pass`. It intentionally includes the `TimedOut` state, so we issue a new request.

The old code gave up this opportunity to retry a timed-out request and rejected the attempt to load the document by returning an error and closing the socket.

The design is to do CheckFileInfo opportunistically on the GET of cool.html. But, unless we have a valid response, we always try again here, when the WS is created. This change shouldn't be risky.

I'll change it to explicitly retry if we haven't tried yet, or we timed out. All other cases go to the error handler.

Reply:
Ah - makes sense =)